### ISSUE TITLE
mxnetR parameter setting

### ISSUE THREAD

**Issue by rflai** [dmlc/mxnet#3373]
Hi All

i am current working on an image classification project by using mxnetR, i faced one question about setting  paramters for this package, especially for the function mx.model.FeedForward.create(). i read couple instructions from different sources.  for parameters wd and initializer in this function, i want to know how to set up these two parameters, what part of the network these two parameter control?

running results for 8 rounds for the following R script

Start training with 1 devices
[1] Train-accuracy=0.19530612244898
[2] Train-accuracy=0.1938
[3] Train-accuracy=0.1948
[4] Train-accuracy=0.1916
[5] Train-accuracy=0.1916
[6] Train-accuracy=0.1916
[7] Train-accuracy=0.1916
[8] Train-accuracy=0.1916

here you are my script (i use the CIFAR-10 pictures as a pilot to set up alexnet, alexnet script was copied from this website)
###### 

library(mxnet)
library(imager)

Lists <- read.table("trainLabels.csv", header = TRUE, sep=",")
Cat <- Lists[(Lists$label == "cat"),]
Dog <- Lists[(Lists$label == "dog"),]
Bird <- Lists[(Lists$label == "bird"),]
Deer <- Lists[(Lists$label == "deer"),]
Horse <- Lists[(Lists$label == "horse"),]
files <- list.files("./train/")

a <- array(0, c(32,32,3,5000))
train.y <- c()

for (i in 1:1000) {
  filename <- paste("",Cat$id[i],".png", sep="")
  if (length(intersect(filename, files)) == 1) {path <- paste("./train/",filename,"", sep="")
  test <- load.image(path)
  a[1:32,1:32,1,(5_(i-1)+1)] <- R(test)
  a[1:32,1:32,2,(5_(i-1)+1)] <- G(test)
  a[1:32,1:32,3,(5_(i-1)+1)] <- B(test)
  train.y[(5_(i-1)+1)] <- 1
  }

  filename <- paste("",Dog$id[i],".png", sep="")
  if (length(intersect(filename, files)) == 1) {path <- paste("./train/",filename,"", sep="")
  test <- load.image(path)
  a[1:32,1:32,1,(5_(i-1)+2)] <- R(test)
  a[1:32,1:32,2,(5_(i-1)+2)] <- G(test)
  a[1:32,1:32,3,(5_(i-1)+2)] <- B(test)
  train.y[(5_(i-1)+2)] <- 2
  }

  filename <- paste("",Bird$id[i],".png", sep="")
  if (length(intersect(filename, files)) == 1) {path <- paste("./train/",filename,"", sep="")
  test <- load.image(path)
  a[1:32,1:32,1,(5_(i-1)+3)] <- R(test)
  a[1:32,1:32,2,(5_(i-1)+3)] <- G(test)
  a[1:32,1:32,3,(5_(i-1)+3)] <- B(test)
  train.y[(5_(i-1)+3)] <- 3
  }

  filename <- paste("",Deer$id[i],".png", sep="")
  if (length(intersect(filename, files)) == 1) {path <- paste("./train/",filename,"", sep="")
  test <- load.image(path)
  a[1:32,1:32,1,(5_(i-1)+4)] <- R(test)
  a[1:32,1:32,2,(5_(i-1)+4)] <- G(test)
  a[1:32,1:32,3,(5_(i-1)+4)] <- B(test)
  train.y[(5_(i-1)+4)] <- 4
  }

  filename <- paste("",Horse$id[i],".png", sep="")
  if (length(intersect(filename, files)) == 1) {path <- paste("./train/",filename,"", sep="")
  test <- load.image(path)
  a[1:32,1:32,1,(5_(i-1)+5)] <- R(test)
  a[1:32,1:32,2,(5_(i-1)+5)] <- G(test)
  a[1:32,1:32,3,(5_(i-1)+5)] <- B(test)
  train.y[(5_(i-1)+5)] <- 5
  }
}
train.array <- round(a, 2)
rm(a)
## \- Define deep learning network, act_type can be relu, sigmoid, softrelu and tanh

input_data <- mx.symbol.Variable(name = "data")
## \- stage 1

conv1 <- mx.symbol.Convolution(data = input_data, kernel = c(8, 8), stride = c(2, 2), num_filter = 96)
relu1 <- mx.symbol.Activation(data = conv1, act_type = "relu")
## \- after stage 1, the data size is (32-8+2*0)/2 + 1 = 13

pool1 <- mx.symbol.Pooling(data = relu1, pool_type = "max", kernel = c(3, 3))
## \- next layer data size: (13-3+2*0)/1 + 1 = 11

lrn1 <- mx.symbol.LRN(data = pool1, alpha = 0.0001, beta = 0.75, knorm = 1, nsize = 5)
## \- stage 2

conv2 <- mx.symbol.Convolution(data = lrn1, kernel = c(5, 5), pad = c(1, 1), num_filter = 96)
relu2 <- mx.symbol.Activation(data = conv2, act_type = "relu")
## \- next layer data size: (11-5+2*1)/1 + 1 = 9

pool2 <- mx.symbol.Pooling(data = relu2, kernel = c(3, 3), pool_type = "max")
## \- next layer data size: (9-3+2*0)/1 + 1 = 7

lrn2 <- mx.symbol.LRN(data = pool2, alpha = 0.0001, beta = 0.75, knorm = 1, nsize = 5)
## \- stage 3

conv3 <- mx.symbol.Convolution(data = lrn2, kernel = c(3, 3), pad = c(1, 1), num_filter = 384)
relu3 <- mx.symbol.Activation(data = conv3, act_type = "relu")
## \- next layer data size: (7-3+2*1)/1 + 1 = 7

conv4 <- mx.symbol.Convolution(data = relu3, kernel = c(3, 3), pad = c(1, 1), num_filter = 384)
relu4 <- mx.symbol.Activation(data = conv4, act_type = "relu")
## \- next layer data size: (7-3+2*1)/1 + 1 = 7

conv5 <- mx.symbol.Convolution(data = relu4, kernel = c(3, 3), pad = c(0, 0), num_filter = 256)
relu5 <- mx.symbol.Activation(data = conv5, act_type = "relu")
## \- next layer data size: (7-3+2*0)/1 + 1 = 5

pool3 <- mx.symbol.Pooling(data = relu5, kernel = c(2, 2), stride = c(1, 1), pool_type = "max")
## \- stage 4

flatten <- mx.symbol.Flatten(data = pool3)
fc1 <- mx.symbol.FullyConnected(data = flatten, num_hidden = 1024)
relu6 <- mx.symbol.Activation(data = fc1, act_type = "relu")
dropout1 <- mx.symbol.Dropout(data = relu6, p = 0.5)
## \- stage 5

fc2 <- mx.symbol.FullyConnected(data = dropout1, num_hidden = 256)
relu7 <- mx.symbol.Activation(data = fc2, act_type = "relu")
dropout2 <- mx.symbol.Dropout(data = relu7, p = 0.5)
## \- stage 6

fc3 <- mx.symbol.FullyConnected(data = dropout2, num_hidden = 5)
softmax <- mx.symbol.SoftmaxOutput(data = fc3, name = 'softmax')

mx.set.seed(0)
device.cpu <- mx.cpu(1)

tic <- proc.time()
model <- mx.model.FeedForward.create(softmax, X=train.array, y=train.y, ctx=device.cpu, 
                                     num.round=10, array.batch.size=100, learning.rate=0.05, 
                                     momentum=0.9, eval.metric=mx.metric.accuracy, 
                                     initializer=mx.init.uniform(0.07),
                                     wd=0.0001,
                                     epoch.end.callback=mx.callback.log.train.metric(100)
                                     )

**Comment by thirdwing** [comment]
@rflai I think the issue is the same with https://github.com/dmlc/mxnet/issues/3625

Please try to set the labels from 0, not 1.

Feel free to reopen if necessary.
